realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java

```diff
@@ -404,6 +405,38 @@ public void equals_cyclicObject() {
     }
 
     @Test
+    public void equals_customMethod() {
+        realm.beginTransaction();
+        CustomMethods cm = realm.createObject(CustomMethods.class);
+        cm.setName("Foo");
+        realm.commitTransaction();
+
+        CustomMethods cm1 = realm.where(CustomMethods.class).findFirst();
+        CustomMethods cm2 = realm.where(CustomMethods.class).findFirst();
+        assertTrue(cm1.equals(cm2));
+
+        realm.beginTransaction();
+        cm1.reverseEquals = true;
+        realm.commitTransaction();
+        assertFalse(cm1.equals(cm2));
+
+        CustomMethods cm3 = new CustomMethods();
```

Comment:
I would move this to its own test method. As a general rule unit tests shouldn't test multiple behaviours. I know we break it in a few places, but here it would take up the same amount of space and be a lot more readable.